api/client/service/update.go

```diff
@@ -357,7 +357,12 @@ func removeItems(
 func updateMounts(flags *pflag.FlagSet, mounts *[]mounttypes.Mount) {
 	if flags.Changed(flagMountAdd) {
 		values := flags.Lookup(flagMountAdd).Value.(*MountOpt).Value()
-		*mounts = append(*mounts, values...)
+
+		exist := findAndReplaceMountData(mounts, values)
```

Comment:
I think, I understood the error for duplicates. It is for input list for update to have duplicate entries. Something similar to updatePorts() function.

Will make necessary changes.